pkg/deviceplugins/vgpu_device_manager.go

```diff
@@ -0,0 +1,499 @@
+package deviceplugins
+
+/* This file was part of the KubeVirt project, copied to this project
+ * to get around private package issues.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 SUSE, LLC.
+ *
+ */
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"kubevirt.io/kubevirt/pkg/util"
+	pluginapi "kubevirt.io/kubevirt/pkg/virt-handler/device-manager/deviceplugin/v1beta1"
+
+	"github.com/harvester/pcidevices/pkg/apis/devices.harvesterhci.io/v1beta1"
+)
+
+const (
+	vgpuPrefix = "MDEV_PCI_RESOURCE"
+)
+
+type VGPUDevicePlugin struct {
+	devs         []*pluginapi.Device
+	server       *grpc.Server
+	socketPath   string
+	stop         <-chan struct{}
+	health       chan deviceHealth
+	devicePath   string
+	resourceName string
+	done         chan struct{}
+	deviceRoot   string
+	initialized  bool
+	lock         *sync.Mutex
+	deregistered chan struct{}
+	starter      *DeviceStarter
+	ctx          context.Context
+}
+
+// Not adding more data to the struct, it's big enough already
+func (dp *VGPUDevicePlugin) GetCount() int {
+	var count int
+	for _, dev := range dp.devs {
+		if dev.Health == pluginapi.Healthy {
+			count++
+		}
+	}
+	logrus.Debugf("found device count %d for plugin %s", count, dp.resourceName)
+	return count
+}
+
+func NewVGPUDevicePlugin(ctx context.Context, vGPUList []string, resourceName string) *VGPUDevicePlugin {
+	serverSock := SocketPath(strings.Replace(resourceName, "/", "-", -1))
+
+	initHandler()
```

Comment:
Can you help explain if initHandler() is needed? IIUC, this function seems not to take effect. Thanks.

Reply:
yeah not needed. legacy stuff that I have now cleaned up.